### DISCUSSION TITLE
Improve display of available actions for alarm_control_panel entities

### DISCUSSION THREAD

**Discussion by nielsfaber** [home-assistant/frontend#11309] [Other feature requests] [answered]
## The request

In the Alarm Panel Lovelace card and the  `ha-dialog` for `alarm_control_panel` entities:

- actions "ARM HOME" and "ARM AWAY" will be displayed if the entity is in DISARMED state
- action "DISARM" will be displayed if the entity is in any other state

The proposed change is that the `supported_features` property shall be used to determine which actions are available.

So in DISARMED state, one or more actions of this set should be displayed:
- ARM AWAY
- ARM HOME
- ARM NIGHT
- ARM CUSTOM BYPASS

Likewise, in all states except DISARMED, one or more of the actions should be displayed:
- DISARM (always supported)
- TRIGGER (only has practical sense in any of the ARMED_* states)

## The alternatives
For the Alarm Panel Lovelace card it is possible to configure which actions are available, so users can make the correct actions show up.
Note that it is even possible to configure actions which are not supported by the entity (so button presses will do nothing).

## Additional information

Supported features for `alarm_control_panel` entities:

```
SUPPORT_ALARM_ARM_HOME = 1
SUPPORT_ALARM_ARM_AWAY = 2
SUPPORT_ALARM_ARM_NIGHT = 4
SUPPORT_ALARM_TRIGGER = 8
SUPPORT_ALARM_ARM_CUSTOM_BYPASS = 16
```

### PS:
Please consider renaming "ARM CUSTOM BYPASS" --> ARM CUSTOM" and "ARMED CUSTOM BYPASS --> ARMED CUSTOM".
The term *bypass* is misleading.
It suggests that sensors are bypassed (ignored) in this state, but this is not necessarily the case.
The behaviour of the sensors per mode is depending on the integration. It is out of scope for this platform.

**Comment by karwosts** [comment, marked as answer]
See https://github.com/home-assistant/frontend/pull/15452
